core/src/main/java/com/scalar/db/storage/dynamo/DynamoAdmin.java

```diff
@@ -1,78 +1,325 @@
+    applicationAutoScalingClientBuilder.credentialsProvider(awsCredentialsProvider);
+    applicationAutoScalingClientBuilder.region(Region.of(config.getContactPoints().get(0)));
+    applicationAutoScalingClient = applicationAutoScalingClientBuilder.build();
+
     namespacePrefix = config.getNamespacePrefix();
     metadataManager = new DynamoTableMetadataManager(client, namespacePrefix);
   }
 
   @VisibleForTesting
   DynamoAdmin(DynamoTableMetadataManager metadataManager, Optional<String> namespacePrefix) {
     client = null;
+    applicationAutoScalingClient = null;
     this.metadataManager = metadataManager;
     this.namespacePrefix = namespacePrefix.map(n -> n + "_");
   }
 
+  @VisibleForTesting
+  DynamoAdmin(
+      DynamoDbClient dynamoDbClient,
+      ApplicationAutoScalingClient applicationAutoScalingClient,
+      DynamoTableMetadataManager metadataManager,
+      Optional<String> namespacePrefix) {
+    this.client = dynamoDbClient;
+    this.applicationAutoScalingClient = applicationAutoScalingClient;
+    this.metadataManager = metadataManager;
+    this.namespacePrefix = namespacePrefix;
+  }
+
   @Override
-  public void createNamespace(String namespace, Map<String, String> options)
-      throws ExecutionException {
-    throw new UnsupportedOperationException("implement later");
+  public void createNamespace(String namespace, Map<String, String> options) {
+    // In Dynamo DB storage, namespace will be added to table name as prefix along with dot
+    // separator.
   }
 
   @Override
   public void createTable(
       String namespace, String table, TableMetadata metadata, Map<String, String> options)
       throws ExecutionException {
-    throw new UnsupportedOperationException("implement later");
+    CreateTableRequest.Builder requestBuilder = CreateTableRequest.builder();
+
+    List<AttributeDefinition> columnsToAttributeDefinitions = new ArrayList<>();
+    makeAttribute(PARTITION_KEY, metadata, columnsToAttributeDefinitions);
+    if (!metadata.getClusteringKeyNames().isEmpty()) {
+      makeAttribute(CLUSTERING_KEY, metadata, columnsToAttributeDefinitions);
+      for (String clusteringKey : metadata.getClusteringKeyNames()) {
+        makeAttribute(clusteringKey, metadata, columnsToAttributeDefinitions);
+      }
+    }
+    if (!metadata.getSecondaryIndexNames().isEmpty()) {
+      for (String secondaryIndex : metadata.getSecondaryIndexNames()) {
+        makeAttribute(secondaryIndex, metadata, columnsToAttributeDefinitions);
+      }
+    }
+    requestBuilder.attributeDefinitions(columnsToAttributeDefinitions);
+
+    // build keys
+    buildPrimaryKey(requestBuilder, metadata);
+
+    // build local indexes that corresponding to clustering keys
+    buildLocalIndexes(namespace, table, requestBuilder, metadata);
+
+    // build secondary indexes
+    long ru = DEFAULT_RU;
+    if (options.containsKey(REQUEST_UNIT)) {
+      ru = Long.parseLong(options.get(REQUEST_UNIT));
+    }
```

Comment:
```suggestion
    long ru = Long.parseLong(options.getOrDefault(REQUEST_UNIT, DEFAULT_RU));
```

Reply:
Apply in [19938a0](https://github.com/scalar-labs/scalardb/pull/301/commits/19938a073c0e7ed103d9d226c6f128cbaa6cfdff)